airbyte-webapp/src/components/connection/CatalogTree/CatalogTree.tsx

```diff
@@ -56,8 +56,13 @@ const CatalogTreeComponent: React.FC<React.PropsWithChildren<CatalogTreeProps>>
 
   const changedStreams = useMemo(
     () =>
-      streams.filter((stream, idx) => {
-        return stream.config?.selected !== initialValues.syncCatalog.streams[idx].config?.selected;
+      streams.filter((stream) => {
+        const matchingInitialValue = initialValues.syncCatalog.streams.find(
+          (initialStream) =>
```

Comment:
We also match by id (which is the stringified index of the item in the array) a little higher up, and in the `BulkEditService`.  TBD whether those should also be changed to use name & namespace as matchers.  I am unsure if either would get called in a case where using the index would be unreliable.

Reply:
I had the same thought as well when I was looking at this. I wonder if we can trust the ID, but in this solution we know we can trust name + namespace as a matching criteria so this is less risky.

Reply:
Do you think it's worth making a ticket to audit other uses of index/stream id?  I'm not _sure_ they're a problem, but I'm not sure they're _not_ a problem.

Reply:
I think it's good to track this in a ticket. If the id proves to be trustworthy, we can simplify a lot of checks in our code.

Reply:
Ope.  I hit merge.  The id is the item's index in the array converted to a string, so it would presumably have the same problems.